### DISCUSSION TITLE
More detailed installation tips for linux

### DISCUSSION THREAD

**Discussion by make-github-pseudonymous-again** [edemaine/coauthor#199] [Installation] [answered]
On linux, the meteor installation script needs tar and gzip. Add line in installation steps about this.

**Comment by make-github-pseudonymous-again** [comment, marked as answer]
Also `git clone` needs git and `meteor npm install` needs python.

**Comment by make-github-pseudonymous-again** [comment]
python2 to be precise and also make.

**Comment by make-github-pseudonymous-again** [comment]
and g++/gcc and sed

**Comment by make-github-pseudonymous-again** [comment]
`meteor mongo` needs ps

**Comment by make-github-pseudonymous-again** [comment]
Also, a weird error can happen for people using MongoDB for the first time on a fresh linux install: [MongoDB needs /etc/default/locale to be filled with stuff](https://github.com/meteor/meteor/issues/4019). Failing to do so will result in the cryptic error

    Unexpected mongo exit code 1. Restarting.
    Unexpected mongo exit code 1. Restarting.
    Unexpected mongo exit code 1. Restarting.
    Can't start Mongo server.

Tell me if I can just edit the README and send you a pull request with all those changes.

**Comment by edemaine** [comment]
Are there UNIX systems without `tar`, `gzip`, `sed`, and `ps`?  I thought these were part of the base install on all known UNIXes, but I guess not.  Which one?  (For example, with a base Ubuntu install, I didn't have to add anything.)

I feel like these are Meteor issues, not directly Coauthor related.  Meteor installation should only be necessary on the development machine, not the deployment server (as `mup` should handle everything needed on there).  So I'm unclear why you're starting with such a bare machine...  (Or are you not going to use `mup`?)

**Comment by make-github-pseudonymous-again** [comment]
Arch Linux

**Comment by edemaine** [comment]
https://www.archlinux.org/groups/i686/base/ ?

**Comment by make-github-pseudonymous-again** [comment]
I agree this is not a common Linux issue. The information could be in the README in a one-liner that says: "Make sure to have tar, gzip, git, python (version 2.x), make, g++, sed and ps installed, as well as your /etc/default/locale properly configured."

**Comment by make-github-pseudonymous-again** [comment]
Ok maybe, this is because I used the base/archlinux docker container.

**Comment by make-github-pseudonymous-again** [comment]
Ok, [the base group should be installed](https://bugs.archlinux.org/task/53919#comment157443).

**Comment by make-github-pseudonymous-again** [comment]
The base group does NOT contain git, python2, make, gcc.

**Comment by jbosboom** [comment]
gcc and make are in [base-devel](https://www.archlinux.org/groups/x86_64/base-devel/); git and python2 are in their eponymous packages.  In general, run `pacman -Fy` to download the files database, then `pacman -Fo <file>` to see which package owns a file (e.g, `pacman -Fo usr/bin/ps`), or `pacman -Fs <file>` to search by filename (e.g., `pacman -Fs ps`).

**Comment by make-github-pseudonymous-again** [comment]
@jbosboom What is the difference with `pacman -Qo` which I used?

**Comment by make-github-pseudonymous-again** [comment]
The `backup.sh` script needs `mongodump` which is found in `mongodb-tools` on Arch (optional dependency of `mongodb` so not automatically installed).

**Comment by edemaine** [comment]
Added documentation about this to `backup.sh` script.  I don't think it's useful to document here how to install Meteor, as that's a Meteor issue, not specific to Coauthor.
